swan-lake/learn/writing-tests.md

```diff
@@ -0,0 +1,345 @@
+---
+layout: ballerina-left-nav-pages-swanlake
+title: Writing Tests
+description: Learn how to use Ballerina's built-in test framework to write tests. The test framework provides a set of
+ annotations and assertions to help write and run tests.
+keywords: ballerina, programming language, testing
+permalink: /swan-lake/learn/writing-tests/
+active: writing-tests
+redirect_from:
+  - /swan-lake/learn/writing-tests
+  - /swan-lake/learn/writing-tests/
+---
+
+# Structuring test code
+
+## Project Structure
+
+
+```
+project-name/
+  Ballerina.toml
+    src/
+      module1/
+        main.bal
+        Module.md 
+        [resources/]   
+        tests/       	  # Module-specific tests
+          main_test.bal   # The test file for main.bal
+          [resources]	  # Resources for the tests
+```
+
+
+
+## Test source files
+
+Unit tests bound to a module need to be placed in a subfolder called _tests/_ within the module. In a standard
+ Ballerina project, a module is mapped to a test suite. All tests within a module’s _tests/_ subfolder are
+  considered to be part of the same test suite.
+
+The test source files could have any name. The test functions are just Ballerina functions that use a special annotation to mark the function as a test. Test functions must be specified with the `@test:Config{}` annotation and there is no restriction on the test function name.
+
+
+## Test Resources
+
+The resources subdirectory found within the _tests/_ directory is meant to contain any files or resources that are
+ exclusively required for testing. You can access the resource files either using the absolute path or using the path relative to the project root.
+
+## Defining a test
+
+The test module provides the necessary annotations to construct a test suite. Therefore, importing the test module is essential in order to write Ballerina tests.
+
+
+```
+import ballerina/test;
+```
+
+
+Once the test module is imported, the following annotation can be used to write a test function.
+
+
+### @test:Config {}
+
+The function specified following the annotation is a test function. This annotation supports the following value fields.
+
+
+
+*   enable: {true | false}: Enable/disable the test. Default: true
+*   before: "&lt;function name>": Name of the function to be run just before the test is run. Default: none
+*   after: "&lt;function name>": Name of the function to be run just after the test is run.
+*   dependsOn: ["&lt;function names>", …]: List of function names on which the test function depends. The order in which the comma-separated list appears has no prominence. In case there needs to be an order, define a sequence of test functions with one pointing to another based on dependency using the dependsOn parameter in each one’s config in order to control the order of test execution.
```

Comment:
```suggestion
*   dependsOn: ["&lt;function names>", …]: List of function names on which the test function depends. The order in which the comma-separated list appears has no prominence. In case there needs to be an order, define a sequence of test functions with one pointing to another based on the dependencies using the `dependsOn` parameter in each one’s config in order to control the order of test execution.
```